### ISSUE TITLE
BLPOP/BRPOP incorrectly handling fractional timeouts

### ISSUE THREAD

**Issue by GoogleCodeExporter** [ashish21590/redis#308]
```
What steps will reproduce the problem?
1. Make sure "foo" "bar" and "baz" are non-existent keys.
2. BLPOP "foo" "bar" "baz" 0.1

What is the expected output? What do you see instead?
The call should block for 0.1 seconds and then return (nil)
Instead, the call blocks indefinitely

What version of the product are you using? On what operating system?
2.0.0-rc4

uname -a
Linux gonzo 2.6.32-24-generic #39-Ubuntu SMP Wed Jul 28 05:14:15 UTC 2010 
x86_64 GNU/Linux


Please provide any additional information below.
I suspect that the timeout parameter is being rounded down to the nearest 
integer.

```

Original issue reported on code.google.com by `tjdzi...@gmail.com` on 17 Aug 2010 at 9:48

**Comment by GoogleCodeExporter** [comment]
```
The timeout is interpreted as an integer, so passing a 0.1 timeout is 
interpreted as a 0 timeout. It is not possible to provide a higher resolution 
timeout for the blocking pop family because Redis is not able to deliver higher 
resolution. The higher resolution timeout may work well when idle, but will 
definitely fail to work when the server is under load. When you would be able 
to specify a higher resolution timeout, you would want Redis to stick to it.

Because this is expected behavior, I'm closing this issue.
```

Original comment by `pcnoordh...@gmail.com` on 18 Aug 2010 at 10:58
- Changed state: **Invalid**

**Comment by GoogleCodeExporter** [comment]
```
Alright, fair enough, and I see that you've updated the documentation. However, 
Redis does not tell you that you're doing something wrong when you specify a 
non-integer time out, so if you specify 0.1, it will silently perform 
unexpected behavior, which is basically a silent failure.
```

Original comment by `tjdzi...@gmail.com` on 18 Aug 2010 at 2:43

**Comment by GoogleCodeExporter** [comment]
```
Sounds reasonable. I've created a patch for this which will probably be merged 
into 2.2. Thanks.
```

Original comment by `pcnoordh...@gmail.com` on 18 Aug 2010 at 4:37

**Comment by GoogleCodeExporter** [comment]
```
This is fixed as of this commit in the 2.2 tree: 
http://github.com/antirez/redis/commit/94364d53b4746e8cd9e3da633162cb1e34f0bdb6
```

Original comment by `pcnoordh...@gmail.com` on 3 Sep 2010 at 8:36
- Changed state: **Verified**
